src/Decoder.cpp

```diff
@@ -106,6 +109,41 @@ namespace nbs {
         }
     }
 
+    Napi::Value Decoder::GetIndices(const Napi::CallbackInfo& info) {
+        Napi::Env env = info.Env();
+
+        auto type_iterators = this->index.getAllTypeIterators();
+        auto indices        = Napi::Array::New(env, type_iterators.size());
+
+        uint64_t index = 0;
+
+        for (const auto& typeIterator : type_iterators) {
+            auto typeSubType = typeIterator.first;
+            auto iterators   = typeIterator.second;
+
+            auto typeIndices = Napi::Object::New(env);
+
+            auto jsType = Napi::Object::New(env);
+            jsType.Set("type", hash::ToJsValue(typeSubType.type, env));
+            jsType.Set("subtype", Napi::Number::New(env, typeSubType.subtype));
+
+            auto timestamps = Napi::Array::New(env, std::distance(iterators.first, iterators.second));
+
+            uint64_t timestampIndex = 0;
+            for (auto it = iterators.first; it != iterators.second; it++) {
+                auto index                   = *it;
+                timestamps[timestampIndex++] = timestamp::ToJsValue(index.item.timestamp, env);
+            }
```

Comment:
use std::transform
```suggestion
            std::transform(iterators.first, iterators.second, timestamps.begin(), [](const IndexItem& i) { return timestamp::ToJsValue(i.item.timestamp, env; });
```

Something like that but I don't think I did it quite correctly :P

Reply:
I don't think this works. The .begin() is an iterator of key/value pairs of the NAPI object, not the array values like you'd get in an std::vector. Maybe I'm just doing something wrong though :P